operate/cli.py

```diff
@@ -663,6 +666,98 @@ async def _stop_service_onchain(request: Request) -> JSONResponse:
             )
         )
 
+    @app.post("/api/services/{service}/onchain/withdraw")
+    @with_retries
+    async def _withdraw_onchain(request: Request) -> JSONResponse:
+        """Withdraw all the funds from a service."""
+        if not operate.service_manager().exists(service=request.path_params["service"]):
+            return service_not_found_error(service=request.path_params["service"])
+        if operate.password is None:
+            return USER_NOT_LOGGED_IN_ERROR
+
+        withdrawal_address = (await request.json()).get("withdrawal_address")
+        if withdrawal_address is None:
+            return JSONResponse(
+                content={"error": "withdrawal_address is required"},
+                status_code=400,
+            )
+
+        try:
+            service_manager = operate.service_manager()
+            service = service_manager.load_or_create(
+                hash=request.path_params["service"]
+            )
+
+            service_manager.terminate_service_on_chain_from_safe(
+                hash=request.path_params["service"],
+                chain_id=service.home_chain_id,
+                is_withdrawing=True,
+            )
+
+            service_manager.drain_service(
+                hash=request.path_params["service"],
+                withdrawal_address=withdrawal_address,
+            )
+
+            # drain OLAS from the master safe
```

Comment:
What if the Safe is needed for some other service?

Reply:
With the current design, it will need to be refilled